test/PushNotificationTests/LRPTests.cpp

```diff
@@ -0,0 +1,183 @@
+
+namespace Test::PushNotifications
+{
+    class LRPTests
+    {
+
+    public:
+        BEGIN_TEST_CLASS(LRPTests)
+            TEST_CLASS_PROPERTY(L"Description", L"Windows App SDK Push Notifications Long Running Process tests")
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA")
+            TEST_CLASS_PROPERTY(L"RunAs:Class", L"RestrictedUser")
+        END_TEST_CLASS()
+
+        TEST_CLASS_SETUP(ClassInit)
+        {
+            try
+            {
+                TP::AddPackage_WindowsAppSDKFramework();           // Installs WASfwk
+                TP::AddPackage_DynamicDependencyDataStore();       // Installs WASmain
+                TP::AddPackage_DynamicDependencyLifetimeManager(); // Installs WASddlm
+                TP::AddPackage_PushNotificationsLongRunningTask(); // Installs the Push Notifications Long Running Process (LRP).
+            }
+            catch (...)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            try
+            {
+                // Remove in reverse order to avoid conflicts between inter-dependent packages.
+                TP::RemovePackage_PushNotificationsLongRunningTask();
+                TP::RemovePackage_DynamicDependencyLifetimeManager();
+                TP::RemovePackage_DynamicDependencyDataStore();
+                TP::RemovePackage_WindowsAppSDKFramework();
+            }
+            catch (...)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_WindowsAppSDKFramework());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_PushNotificationsLongRunningTask());
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_WindowsAppSDKFramework());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_PushNotificationsLongRunningTask());
+
+            return true;
+        }
+
+        TEST_METHOD(LaunchLRP_FromCoCreateInstance)
+        {
+            VERIFY_SUCCEEDED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+            auto scopeExit = wil::scope_exit(
+                [&]() { VERIFY_NO_THROW(CoUninitialize()); });
+
+            VERIFY_NO_THROW(winrt::create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL));
+
+            // Poll the app status every second. It should be alive for 5 seconds.
+            for (int i = 0; i < 5; i++)
```

Comment:
May be best by adding a one second extra buffer since you are polling? So make it 6?

Reply:
What's the intention of it?